fink_science/anomaly_detection/processor.py

```diff
@@ -51,135 +49,105 @@ def anomaly_score(self, data_g, data_r):
-    if not (os.path.exists(r_model_path) and os.path.exists(g_model_path)):
-        with zipfile.ZipFile(f"{model_path}/anomaly_detection_forest.zip", 'r') as zip_ref:
-            zip_ref.extractall(tmp_path)
-    if not (os.path.exists(g_model_path_AAD) and os.path.exists(r_model_path_AAD)):
-        with zipfile.ZipFile(f"{model_path}/anomaly_detection_forest_AAD.zip", 'r') as zip_ref:
-            zip_ref.extractall(tmp_path)
-
-
-class WrapInferenceSession:
-    """
-    The class is an additional wrapper over InferenceSession
-    to solve the pyspark serialisation problem
-
-    https://github.com/microsoft/onnxruntime/pull/800#issuecomment-844326099
-    """
-    def __init__(self, onnx_bytes):
-        self.sess = rt.InferenceSession(onnx_bytes.SerializeToString())
-        self.onnx_bytes = onnx_bytes
-
-    def run(self, *args):
-        return self.sess.run(*args)
-
-    def __getstate__(self):
-        return {'onnx_bytes': self.onnx_bytes}
-
-    def __setstate__(self, values):
-        self.onnx_bytes = values['onnx_bytes']
-        self.sess = rt.InferenceSession(self.onnx_bytes.SerializeToString())
-
+    Parameters
+    ----------
+    lc_features: Spark Map
+        Dict of dicts of floats. Keys of first dict - filters (fid), keys of inner dicts - names of features.
 
-forest_r = WrapInferenceSession(load(r_model_path))
-forest_g = WrapInferenceSession(load(g_model_path))
-forest_r_AAD = WrapInferenceSession(load(r_model_path_AAD))
-forest_g_AAD = WrapInferenceSession(load(g_model_path_AAD))
+    Returns
+    ----------
+    out: float
+        Anomaly score
 
+    Examples
+    ---------
+    >>> from fink_utils.spark.utils import concat_col
+    >>> from pyspark.sql import functions as F
+    >>> from fink_science.ad_features.processor import extract_features_ad
 
-r_means = pd.read_csv(f"{model_path}/r_means.csv", header=None, index_col=0, squeeze=True)
-g_means = pd.read_csv(f"{model_path}/g_means.csv", header=None, index_col=0, squeeze=True)
+    >>> df = spark.read.load(ztf_alert_sample)
 
-model = TwoBandModel(forest_g, forest_r)
-model_AAD = TwoBandModel(forest_g_AAD, forest_r_AAD)
+    # Required alert columns, concatenated with historical data
+    >>> what = ['magpsf', 'jd', 'sigmapsf', 'fid', 'distnr', 'magnr', 'sigmagnr', 'isdiffpos']
+    >>> prefix = 'c'
+    >>> what_prefix = [prefix + i for i in what]
+    >>> for colname in what:
+    ...    df = concat_col(df, colname, prefix=prefix)
 
+    >>> cols = ['cmagpsf', 'cjd', 'csigmapsf', 'cfid', 'objectId', 'cdistnr', 'cmagnr', 'csigmagnr', 'cisdiffpos']
+    >>> df = df.withColumn('lc_features', extract_features_ad(*cols))
+    >>> df = df.withColumn("anomaly_score", anomaly_score("lc_features"))
 
-def anomaly_score(lc_features, model_type='AADForest'):
-    @udf(returnType=DoubleType())
-    def anomaly_score(lc_features) -> float:
-        """ Returns anomaly score for an observation
+    >>> df.filter(df["anomaly_score"] < -0.5).count()
+    0
```

Comment:
Use another threshold -- having 0 classification is not very telling. What about `df.filter(df["anomaly_score"] < -0.1).count()`